### ISSUE TITLE
Would you accept a PR for CockroachDB as a backend?

### ISSUE THREAD

**Issue by sean-** [gravitational/teleport#1742]
If a backend appeared for cockroachdb, would a PR be accepted?

**Comment by klizhentas** [comment]
What kind of deps does cockroach db client have?

**Comment by sean-** [comment]
@klizhentas It speaks PostgreSQL.  I would bring in `jackc/pgx` as a library.

**Comment by klizhentas** [comment]
how does it deal with expiration and TTL?

**Comment by sean-** [comment]
@klizhentas The schema would need to have a `valid_until` and remove dead rows.

**Comment by klizhentas** [comment]
@sean- Sorry for the long response - we were busy rolling 2.5.0 out.

TLDR: We will work with you on this using plugin system as a way to add support.

We don't mind adding support to cockroachdb at all, however we would like to use this as an opportunity to split community supported backends with core backends, as the core code base is a subject to security audits.

To make this possible, we would like to work with you on adding this as a first backend in a community repository.

Essentially we would have to add plugin system to teleport to achieve this.

We were looking at using plugin package:

https://golang.org/pkg/plugin/

I would need to do more research at the prior art of using this feature to understand all the possible caveats. Let us write the design of the plugin in this ticket and and track it here.

**Comment by fire** [comment]
I am also interested in this. Wanted to know if this is blocked on anything the community can provide.

**Comment by klizhentas** [comment]
I'm assuming CockroachDB wont' be that far away from https://github.com/gravitational/teleport/blob/master/lib/backend/lite/lite.go so if anyone is interested giving it a go, we would review the PR to the backend/. Just start with a design proposal of how this would look like and we will provide the feedback

**Comment by zmb3** [comment]
Closing as we added support for CockroachDB in Teleport 8
